docs/routes.md

```diff
@@ -0,0 +1,268 @@
+
+Example: `GET /components`
+```json
+[
+  "text",
+  "image",
+  "paragraph",
+  "article"
+]
+```
+
+`GET /components/:name/instances` will return a list of the available instances within a particular component type.
+
+Example: `GET /components/text/instances`
+```json
+[
+  "/components/text/instances/abc",
+  "/components/text/instances/def",
+  "/components/text/instances/def@published"
+]
+```
+
+### Component Data
+
+You can grab the default data for a component, data for a specific instance, or even data for a specific version (of an instance). GETs and PUTs to these endpoints behave as you would expect from a RESTful API (they return what you send them). GETs and PUTs to the `.json` and `.html` extensions also work, but they'll return the composed (i.e. a component and its children) JSON and HTML, respectively.
+
+### Modifying components
+`GET /components/:name[/instances/:id][@:version[.:extension]]` will return a component.  The form the data is based on the extension (.html, .json, .yaml), or the Accepts header of the request.  
+
+Requesting data without a version will return the latest saved version.  Some versions have special rules (see [Propagating Versions](#propagating-versions)), but any other version name can be used to specially tag any particular version.
+
+`PUT /components/:name[/instances/:id][@version]` will save the data such that `GET`ing the same URI will return exactly what was put there.
+
+### Component Logic
+
+Components may have pre-render and pre-save hooks, which do logic before rendering and saving, respectively. To add these, create a `model.js` file in the component's folder, and export `render` and/or `save` methods. Both of these methods should return objects with the component data, which may be wrapped in promises.
+
+_Note:_ It is much better to do component logic when saving rather than every time it wants to render.
+
+```js
+module.exports.render = function (ref, data, locals) {
+  // do logic
+  return data; // will be sent to the template
+}
+
+module.exports.save = function (ref, data, locals) {
+  // do logic
+  return data; // will be sent to the database
+}
+```
+
+You may pass `componenthooks=false` as a query param when doing API calls if you want to completely bypass the `model.js` (e.g. if you've already run through the logic client-side).
+
+### Legacy Server Logic
+
+This feature is deprecated as of amphora v2.11.0 and will be removed in the next major version (it is supplanted by the isomorphic `model.js` files). Legacy server-side logic lives in a `server.js` or `index.js` file in the component's folder, and has a few differences from the `model.js`:
+
+* the default exported function (run on `GET`) does not automatically receive data from the database and must fetch it manually
+* the `put()` should return a database operation or array of operations (which may be wrapped in promises), rather than the component data itself
```

Comment:
```suggestion
* The `put()` should return a database operation or array of operations (which may be wrapped in promises), rather than the component data itself
```